lib/api/apiUtils/object/setPartRanges.js

```diff
@@ -37,7 +37,7 @@ function setPartRanges(dataLocations, outerRange) {
     const max = end - begin + 1;
     let total = 0;
     for (let i = 0; i < dataLocations.length; i++) {
-        if (total >= max) {
+        if (total >= (max - 1)) {
```

Comment:
This seems wrong, the problem is that we don't specify clearly the convention of whether `end` is inclusive or exclusive, ending up having different callers using different conventions. See e.g. https://github.com/scality/S3/blob/master/lib/api/objectGet.js#L188, where the part end boundary is exclusive but `byteRange` is inclusive. I haven't gone through other callers, my gut feeling is that it's meant to be inclusive (hence the comment about `+1` in `setPartRanges`).

I suggest we should go through those callers and fix if they don't follow the convention (hopefully only the part get is in this case but we never know), and document whether it's inclusive or exclusive in JSDoc of `setPartRanges` to avoid future mistakes.

Reply:
Btw, thanks @alexandre-merle for nailing down the issue :+1: